### DISCUSSION TITLE
v1.107.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#10758] [Announcements]
# v1.107.0

## Highlights

Welcome to version `v1.107.0` of Immich. This release includes many bug fixes, new translations from many new languages, and enhancements to the recent new features in the last release, as well as a change in the default timeline display on the mobile app. Check out some of the highlights below:

- Enhancement to deduplication UI - bulk resolve duplicates, view duplicates in fullscreen.
- Display all local assets on the mobile timeline by default.
- Custom request headers for the mobile app.
- Facial recognition now recognizes more faces.

### Deduplication UI Enhancement

Thank you all for your feedback! The deduplication UI now has two bulk actions that apply to all duplicate groups: deduplicating assets automatically and keeping all duplicates. This makes managing thousands of duplication groups much more efficient.

Additionally, you can now view the duplicates in fullscreen mode, making it easier to decide which asset to keep. We also added the status for favorite photos to the card as well.

![image](https://github.com/immich-app/immich/assets/27055614/36ba9e42-2615-47f9-ac23-932815a72abd)

### Display all local assets on the mobile app timeline

Starting from this release, all local assets will be displayed on the main timeline. This is the first step in refactoring the upload mechanism, simplifying the logic, and introducing similar behaviors to a default gallery app on your phone.

With this change, you can now view, manage, and select a certain group of photos you want to upload to Immich directly from the timeline without the need to select them first on the backup page. The album selection for backup can be used for mass backup selection like before.

### Custom request headers for the mobile app

<img src="https://github.com/immich-app/immich/assets/27055614/eb2fe7c1-edd6-44b4-9bb7-65dc033fa680" width="400"/>

Thanks to @matejkramny, you can now add custom headers to all the requests from the mobile app. This mechanism will help many users who use [Cloudflare's Zero Trust Service](https://developers.cloudflare.com/cloudflare-one/identity/service-tokens/)

You can access this setting from the app `Settings > Advanced > Custom proxy headers`

### Facial recognition enhancement

Due to a change in how recognition data is indexed, facial recognition is now more likely to recognize a face as a particular person. This means more assets will have complete facial recognition and the initial clustering will produce fewer duplicate people. Facial recognition is also faster as part of this change.

Clicking the *Missing* button for Facial Recognition from the job panel will make this change apply to existing assets. (This is also run periodically, so it will soon take effect even without explicitly running facial recognition.)

### 🗄️ Server
* fix(server): oauth linking error message by @jrasm91 in https://github.com/immich-app/immich/pull/10287
* fix(server): headers already send by @jrasm91 in https://github.com/immich-app/immich/pull/10289
* fix(deps): update typescript-projects by @renovate in https://github.com/immich-app/immich/pull/10105
* feat: user preferences for archive download size by @danieldietzler in https://github.com/immich-app/immich/pull/10296
* fix(deps): update dependency exiftool-vendored to v26.2.0 by @renovate in https://github.com/immich-app/immich/pull/10102
* refactor(server): partner ids by @jrasm91 in https://github.com/immich-app/immich/pull/10321
* fix(server): include trashed assets in forced thumbnail generation by @michelheusschen in https://github.com/immich-app/immich/pull/10389
* fix(server): orientation handling for person thumbnails by @mertalev in https://github.com/immich-app/immich/pull/10382
* feat(server): separate face search relation by @mertalev in https://github.com/immich-app/immich/pull/10371
* fix(server): include archived assets in forced thumbnail generation by @michelheusschen in https://github.com/immich-app/immich/pull/10409
* chore(server): update exiftool and migrate off deprecated method signatures by @stephen304 in https://github.com/immich-app/immich/pull/10367
* fix(server): consider all I-frames for video thumbnails by @mertalev in https://github.com/immich-app/immich/pull/10471
* fix(server): let thumbnail generation fail on error by @mertalev in https://github.com/immich-app/immich/pull/10479
* fix(server): video thumbnail generation failing with single i-frame by @mertalev in https://github.com/immich-app/immich/pull/10477
* chore(deps): bump braces from 3.0.2 to 3.0.3 in /server by @dependabot in https://github.com/immich-app/immich/pull/10487
* chore(deps): update base-image to v20240618 (major) by @renovate in https://github.com/immich-app/immich/pull/10457
* fix(server): library refresh not checking trashed assets by @mertalev in https://github.com/immich-app/immich/pull/10495
* fix(server): reindex after changing to a model with a different dimension size by @mertalev in https://github.com/immich-app/immich/pull/10496
* fix(server): allow library id to be null in metadata search by @danieldietzler in https://github.com/immich-app/immich/pull/10512
* fix(server): `/places` entries sometimes not ordered alphabetically by @mertalev in https://github.com/immich-app/immich/pull/10514
* fix(deps): update typescript-projects by @renovate in https://github.com/immich-app/immich/pull/10611
* fix(deps): update typescript-projects by @renovate in https://github.com/immich-app/immich/pull/10616
* chore(deps): use full semver docker tag for node images by @jrasm91 in https://github.com/immich-app/immich/pull/10613
* chore(deps): pin dependencies by @renovate in https://github.com/immich-app/immich/pull/10618
* chore(deps): update dependency @types/node to ^20.14.6 by @renovate in https://github.com/immich-app/immich/pull/10627
* chore: update docker node alpine versions to 3.20 by @zackpollard in https://github.com/immich-app/immich/pull/10621
* chore(deps): update base-image to v20240625 (major) by @renovate in https://github.com/immich-app/immich/pull/10620
* chore: bump node docker versions by @zackpollard in https://github.com/immich-app/immich/pull/10629
* chore(deps): pin node.js to 0ccc08f by @renovate in https://github.com/immich-app/immich/pull/10628
* chore(deps): update dependency typescript to v5.5.2 by @renovate in https://github.com/immich-app/immich/pull/10633
* chore: build metadata by @jrasm91 in https://github.com/immich-app/immich/pull/10612
* fix(server): user delete with stacked assets by @jrasm91 in https://github.com/immich-app/immich/pull/10642
* fix(server): about info version by @michelheusschen in https://github.com/immich-app/immich/pull/10659
* fix(server): live photo relation by @jrasm91 in https://github.com/immich-app/immich/pull/10637
* refactor(server): event emits by @jrasm91 in https://github.com/immich-app/immich/pull/10648
* fix(web): new feature photo by @martabal in https://github.com/immich-app/immich/pull/9443
* refactor: move /server-info endpoints to /server by @zackpollard in https://github.com/immich-app/immich/pull/10677
* fix: album remove asset bug by @jrasm91 in https://github.com/immich-app/immich/pull/10687
* fix(server): show partners assets on timeline without permission by @alextran1502 in https://github.com/immich-app/immich/pull/10705
* chore: Lower default duplicate detection distance by @bo0tzz in https://github.com/immich-app/immich/pull/10703
* feat(server): user and server license endpoints by @zackpollard in https://github.com/immich-app/immich/pull/10682
### 📱 Mobile
* chore(deps): update dependency flutter to v3.22.2 by @renovate in https://github.com/immich-app/immich/pull/10158
* fix(app-bar): remove safe area of the app bar in photos page by @RanKKI in https://github.com/immich-app/immich/pull/10340
* fix(mobile): translation for memory title by @RanKKI in https://github.com/immich-app/immich/pull/10324
* fix(mobile): search page by @RanKKI in https://github.com/immich-app/immich/pull/10385
* fix(mobile): my location button on maps not visible due to bottom padding by @RanKKI in https://github.com/immich-app/immich/pull/10384
* fix(mobile): asset description is not shown on the sheet when opened for the first time by @RanKKI in https://github.com/immich-app/immich/pull/10377
* fix(mobile): upgrade maplibre_gl package to fix issue with crash in ios7.4 above simulator by @Mujhtech in https://github.com/immich-app/immich/pull/10182
* feat(mobile): render assets on device by default by @alextran1502 in https://github.com/immich-app/immich/pull/10470
* chore(mobile): remove exclude album mechanism for backup by @alextran1502 in https://github.com/immich-app/immich/pull/10552
* fix(mobile): inconsistent thumbnail's label by @RanKKI in https://github.com/immich-app/immich/pull/10589
* feat(mobile): add additional request headers by @matejkramny in https://github.com/immich-app/immich/pull/10588
* fix(mobile): asset state remain in gallery view after being deleted by @alextran1502 in https://github.com/immich-app/immich/pull/10603
* chore(trans): add zh-CN translations for custom proxy headers by @Triple-Z in https://github.com/immich-app/immich/pull/10660
* chore(mobile): Revert "remove exclude album mechanism for backup (#10552)" by @alextran1502 in https://github.com/immich-app/immich/pull/10686
### 🖥️ Web
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/10285
* chore(web): standardize settings labels by @ben-basten in https://github.com/immich-app/immich/pull/10303
* refactor(web): byte unit utils by @danieldietzler in https://github.com/immich-app/immich/pull/10332
* feat(web): add cover images to individual shares by @Snowknight26 in https://github.com/immich-app/immich/pull/9988
* fix(web): prevent new uploads from temporarily showing in trash by @michelheusschen in https://github.com/immich-app/immich/pull/10348
* fix(web): allow sending test email when using config file by @michelheusschen in https://github.com/immich-app/immich/pull/10351
* feat(web): full screen view for duplicates by @michelheusschen in https://github.com/immich-app/immich/pull/10346
* fix(web): play video muted when blocked by browser by @michelheusschen in https://github.com/immich-app/immich/pull/10383
* fix(web): update avatar color immediately by @michelheusschen in https://github.com/immich-app/immich/pull/10393
* fix(web): selecting shared link expiration by @michelheusschen in https://github.com/immich-app/immich/pull/10437
* fix(web): confirm button is disabled if two dialogs are shown subsequently by @alextran1502 in https://github.com/immich-app/immich/pull/10440
* fix(web): Update prompt by @aviv926 in https://github.com/immich-app/immich/pull/10237
* feat(web): keyboard accessible context menus by @ben-basten in https://github.com/immich-app/immich/pull/10017
* fix(web): comment send button by @waclaw66 in https://github.com/immich-app/immich/pull/10453
* feat(web): display original heif images for safari by @mertalev in https://github.com/immich-app/immich/pull/10478
* feat(web): bulk deduplicate by @mertalev in https://github.com/immich-app/immich/pull/10448
* feat(web): duplicate ui tweaks by @michelheusschen in https://github.com/immich-app/immich/pull/10506
* chore(web): context menu improvements by @ben-basten in https://github.com/immich-app/immich/pull/10475
* feat(web): Adds sort order from album to shared album. by @erathmus in https://github.com/immich-app/immich/pull/10528
* fix(web): six digit year input by @michelheusschen in https://github.com/immich-app/immich/pull/10517
* feat(web): translations containing html by @michelheusschen in https://github.com/immich-app/immich/pull/10491
* fix(web): missing translations by @michelheusschen in https://github.com/immich-app/immich/pull/10504
* fix(web): FormatMessage development keys by @michelheusschen in https://github.com/immich-app/immich/pull/10536
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/10304
* chore(web): another missing translations by @waclaw66 in https://github.com/immich-app/immich/pull/10274
* feat: more languages by @jrasm91 in https://github.com/immich-app/immich/pull/10595
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/10593
* chore(web): translation finetuning by @waclaw66 in https://github.com/immich-app/immich/pull/10601
* chore: merge weblate by @jrasm91 in https://github.com/immich-app/immich/pull/10604
* feat(web): link router by @jrasm91 in https://github.com/immich-app/immich/pull/10644
* fix(web): match storage_template_migration_job with storage_template_migration by @MiguelDomingues in https://github.com/immich-app/immich/pull/10662
* feat(web): show favorite icon in duplicate asset by @Pascal-So in https://github.com/immich-app/immich/pull/10688
* feat(web): use websocket to update the feature photo by @martabal in https://github.com/immich-app/immich/pull/10683
* fix(web): i18n race condition in load function by @michelheusschen in https://github.com/immich-app/immich/pull/10693
* feat(web): add more translations by @michelheusschen in https://github.com/immich-app/immich/pull/10700
* fix(web): shared link card by @michelheusschen in https://github.com/immich-app/immich/pull/10702
* fix(web): Sort timezones in assets settings by offset by @WazWazTheDeveloper in https://github.com/immich-app/immich/pull/10697
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/10605
* fix(web): remove black bezels + better integrate ActivityStatus by @pyorot in https://github.com/immich-app/immich/pull/10667
* feat(web): html tags inside plural and select messages by @michelheusschen in https://github.com/immich-app/immich/pull/10696
* fix(web): suppress album upload notification by @alextran1502 in https://github.com/immich-app/immich/pull/10717
* fix(web): activity status padding regression by @alextran1502 in https://github.com/immich-app/immich/pull/10734
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/10729
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/10742
* feat(web): add keyboard shortcuts to duplicates utility by @tastelessjolt in https://github.com/immich-app/immich/pull/10736
* fix(web): scroll jank on memories page by @michelheusschen in https://github.com/immich-app/immich/pull/10752
### 🧠 Machine Learning
* chore(deps): update machine-learning by @renovate in https://github.com/immich-app/immich/pull/10446
* chore(deps): update mambaorg/micromamba:bookworm-slim docker digest to b17c9b1 by @renovate in https://github.com/immich-app/immich/pull/10465
* chore(ml): support python 3.12 by @mertalev in https://github.com/immich-app/immich/pull/10481
* fix(ml): limit load retries by @mertalev in https://github.com/immich-app/immich/pull/10494
* refactor(ml): model sessions by @mertalev in https://github.com/immich-app/immich/pull/10559
* fix(deps): update machine-learning by @renovate in https://github.com/immich-app/immich/pull/10610
* chore(deps): update mambaorg/micromamba:bookworm-slim docker digest to 333f759 by @renovate in https://github.com/immich-app/immich/pull/10631
### ⚡ CLI
* chore(deps): update node.js to eb17a08 - autoclosed by @renovate in https://github.com/immich-app/immich/pull/10098
* fix(cli): handle patterns correctly on Windows by @fky2015 in https://github.com/immich-app/immich/pull/10430
* fix(cli): fix broken --album on Windows. by @fky2015 in https://github.com/immich-app/immich/pull/10626
### 📓 Documentation
* chore: fix typo in jobs-workers.md by @YodaDaCoda in https://github.com/immich-app/immich/pull/10302
* chore: update discord links by @danieldietzler in https://github.com/immich-app/immich/pull/10301
* fix(doc): literal translation & missing parts by @Rooyca in https://github.com/immich-app/immich/pull/10295
* docs: brief instructions to recover from corruption by @mmomjian in https://github.com/immich-app/immich/pull/10319
* chore: update Unraid Docker-Compose documentation to reflect missing healthcheck start_interval parameter from Docker Engine v24.0.9. by @flemesre in https://github.com/immich-app/immich/pull/10406
* docs: add info about translations and weblate by @jrasm91 in https://github.com/immich-app/immich/pull/10591
* docs: add private photos to roadmap by @jrasm91 in https://github.com/immich-app/immich/pull/10599
* docs: Add new community guide to access Immich with a custom domain by @ppr88 in https://github.com/immich-app/immich/pull/10638
* fix(docs): restart DB backup container by @mmomjian in https://github.com/immich-app/immich/pull/10671
* chore(doc): small punctuation fix backup-and-restore.md by @TheDidek in https://github.com/immich-app/immich/pull/10704
* docs: blog post July 2024 update by @alextran1502 in https://github.com/immich-app/immich/pull/10737
### 🔨 Maintenance
* chore(deps): bump docker/build-push-action from 5.4.0 to 6.0.0 by @dependabot in https://github.com/immich-app/immich/pull/10433
* chore(deps): update dependency exiftool-vendored to v27 by @renovate in https://github.com/immich-app/immich/pull/10447
* chore(deps): update typescript-projects by @renovate in https://github.com/immich-app/immich/pull/10445
* chore(deps): update prom/prometheus docker digest to 075b1ba by @renovate in https://github.com/immich-app/immich/pull/10484
* chore(deps): bump ws and engine.io-client in /e2e by @dependabot in https://github.com/immich-app/immich/pull/10488
* chore(deps): bump docker/build-push-action from 6.0.0 to 6.0.1 by @dependabot in https://github.com/immich-app/immich/pull/10483
* chore(deps): update dependency eslint-plugin-unicorn to v54 by @renovate in https://github.com/immich-app/immich/pull/10486
* fix(deployment): Postgres healthcheck, add username to pg_isready by @mmomjian in https://github.com/immich-app/immich/pull/10221
* chore(deps): update redis:6.2-alpine docker digest to 328fe6a by @renovate in https://github.com/immich-app/immich/pull/10516
* chore(deps): update docker.io/redis:6.2-alpine docker digest to 328fe6a by @renovate in https://github.com/immich-app/immich/pull/10515
* chore(deps): bump docker/build-push-action from 6.0.1 to 6.1.0 by @dependabot in https://github.com/immich-app/immich/pull/10522
* chore: use immich.app email for security reports by @jrasm91 in https://github.com/immich-app/immich/pull/10594
* chore(deps): update dependency prettier-plugin-svelte to v3.2.5 by @renovate in https://github.com/immich-app/immich/pull/10623
* chore(deps): update terraform cloudflare to v4.35.0 by @renovate in https://github.com/immich-app/immich/pull/10420
* chore: add renovate makefile command for testing renovate changes by @zackpollard in https://github.com/immich-app/immich/pull/10630
* chore: better auto labels by @jrasm91 in https://github.com/immich-app/immich/pull/10632
* chore(deps): update dependency @types/node to ^20.14.7 by @renovate in https://github.com/immich-app/immich/pull/10635
* chore(deps): update grafana/grafana docker tag to v11.1.0 by @renovate in https://github.com/immich-app/immich/pull/10679
* chore(deps): update terraform cloudflare to v4.36.0 by @renovate in https://github.com/immich-app/immich/pull/10718
* chore(deps): bump docker/build-push-action from 6.1.0 to 6.2.0 by @dependabot in https://github.com/immich-app/immich/pull/10655
* chore(deps): update node by @renovate in https://github.com/immich-app/immich/pull/10741
* fix(deps): update typescript-projects by @renovate in https://github.com/immich-app/immich/pull/10744
* chore(deps): update docker/build-push-action action to v6.2.0 by @renovate in https://github.com/immich-app/immich/pull/10745
### Other changes
* chore(docs): update Korean README by @bunubbv in https://github.com/immich-app/immich/pull/10462
* ci: use .nvmrc for node-setup node-version in github actions by @zackpollard in https://github.com/immich-app/immich/pull/10619
* ci: group docker node and npm node updates together by @zackpollard in https://github.com/immich-app/immich/pull/10625
* chore(deps): upgrade @testing-library/svelte by @benmccann in https://github.com/immich-app/immich/pull/10690
* docs: sync Chinese README with the official English version by @indam in https://github.com/immich-app/immich/pull/10724
* docs: Translations update by @alextran1502 in https://github.com/immich-app/immich/pull/10730

## New Contributors
* @YodaDaCoda made their first contribution in https://github.com/immich-app/immich/pull/10302
* @Rooyca made their first contribution in https://github.com/immich-app/immich/pull/10295
* @RanKKI made their first contribution in https://github.com/immich-app/immich/pull/10340
* @flemesre made their first contribution in https://github.com/immich-app/immich/pull/10406
* @Mujhtech made their first contribution in https://github.com/immich-app/immich/pull/10182
* @erathmus made their first contribution in https://github.com/immich-app/immich/pull/10528
* @fky2015 made their first contribution in https://github.com/immich-app/immich/pull/10430
* @ppr88 made their first contribution in https://github.com/immich-app/immich/pull/10638
* @matejkramny made their first contribution in https://github.com/immich-app/immich/pull/10588
* @Triple-Z made their first contribution in https://github.com/immich-app/immich/pull/10660
* @MiguelDomingues made their first contribution in https://github.com/immich-app/immich/pull/10662
* @Pascal-So made their first contribution in https://github.com/immich-app/immich/pull/10688
* @TheDidek made their first contribution in https://github.com/immich-app/immich/pull/10704
* @WazWazTheDeveloper made their first contribution in https://github.com/immich-app/immich/pull/10697
* @pyorot made their first contribution in https://github.com/immich-app/immich/pull/10667
* @tastelessjolt made their first contribution in https://github.com/immich-app/immich/pull/10736

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.106.4...v1.107.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.107.0'>v1.107.0</a>.</em>

**Comment by Triple-Z** [comment]
great to hear this version finally supports the custom headers :tada: !

**Comment by JordyEGNL** [comment]
It now shows ALL IMAGES on your timeline, even folders that are not backed up to Immich. In my case all random downloaded images and images from Whatsapp are also shown, will there be an option to hide those images in the timeline?

**Reply by stephen304** [reply]
Yeah technically it now emulates my stock gallery app, but I kinda liked that google photos only shows your camera roll in the timeline and has everything outside of the "Camera" folder available in the library section by folder. Currently immich's library tab only shows "Camera" under the "Albums on device" (I don't know why other local-only asset folders aren't shown there like screenshots, downloads, etc), my preference would be some way to have local-only non-camera assets show up there under their respective folder in the library tab instead of merging them into the timeline.

**Reply by alextran1502** [reply]
Yeah, at first, we wanted to show only the device's camera roll, but iOS doesn't have a camera roll album, so we are just showing everything now. 

I might work on a mechanism not to include a specific album as default, though

**Reply by JordyEGNL** [reply]
> Yeah, at first, we wanted to show only the device's camera roll, but iOS doesn't have a camera roll album, so we are just showing everything now. 
> 
> I might work on a mechanism not to include a specific album as default, though

Why not only show the backup images, and show the rest under the library button in the folders

**Reply by alextran1502** [reply]
> > Yeah, at first, we wanted to show only the device's camera roll, but iOS doesn't have a camera roll album, so we are just showing everything now.
> > I might work on a mechanism not to include a specific album as default, though
> 
> Why not only show the backup images, and show the rest under the library button in the folders

As mentioned in the release note, it is an effort to simplify the logic of the timeline and backup mechanism to prepare for future work on these topics. Each of us have different way of using the main timeline, we have to put in the effort to standardize it to most people

**Reply by PsychotherapistSam** [reply]
I agree with the previous comments as well. A toggle / setting or something similar so that only the albums that have been selected for the backup are shown would be very nice :)

**Reply by JordyEGNL** [reply]
> As mentioned in the release note, it is an effort to simplify the logic of the timeline and backup mechanism to prepare for future work on these topics. Each of us have different way of using the main timeline, we have to put in the effort to standardize it to most people

Thanks for the clarification, hopefully there will be a toggle in a future release :)

**Reply by Sjustein** [reply]
I don't primarily use immich to view my gallery, but to view my photo collection, kind of like a mirror of the web interface. A toggle would be relatively simple and cater to the use cases of both groups of people. I would love to use immich instead of google photo's as my gallery app, but I would need separate timeline and 'library/other albums' tabs (like in google photo's).

**Reply by elversoee** [reply]
Nooooo....
I do not want screenshot and the like on my timeline only assets that are choosen to Be backed Up/on the immich server...we have other gallery apps for that kind of things. (on android.. not using IOS)

Hope to see a feature to make the timeline  behave like 1.106 how ever the Immich Team thinks is the right solution best of luck

**Reply by Kadz93** [reply]
I'm sorry but implementing such a breaking change that fu*s with everyone that has used the app, without a possibility to turn this behaviour off is bonkers.
If you want people to pay you should think about the user and not your "first step of refactoring". You could easily wait and implement the toggle.
Sorry but really really unprofessional.

**Reply by alextran1502** [reply]
Thanks for the feedback, we will push out a hotfix release to revert this behavior and work on a proper implementation in the next release 

<img src="https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExdXY3ZjE0OHQwNmpkc2NnNDh3bjhvb3hxdmlwZ3Q0NnluemR0bWN3ZiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/gPfluBR7dYRWrIdS7j/giphy.gif"/>

**Reply by JordyEGNL** [reply]
> Thanks for the feedback, we will push out a hotfix release to revert this behavior and work on a proper implementation in the next release 
> 
> <img src="https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExdXY3ZjE0OHQwNmpkc2NnNDh3bjhvb3hxdmlwZ3Q0NnluemR0bWN3ZiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/gPfluBR7dYRWrIdS7j/giphy.gif"/>

One of the main reasons I like Immich, listening to community feedback and using that to make Immich better with every update 🤩

**Comment by molnarti** [comment]
for the deduplication i am still missing a few feature, especially when using external libraries:
- indication of the files location - without this information, when i have two folders with the same content, instead of deleting one of the folders with all their files i can end up with keeping both folders, each with half of the files
- marking folders as ignored for deduplication, e.g. to avoid situations when raw and post-processed versions of the same picture are shown as duplicates

**Reply by stephen304** [reply]
I haven't tried deduplication yet but this sounds like it might be useful to allow stacking from that UI too, so that if you get an original, a copy, and an edit, you can see that the copy had some metadata removed so you mark that for deletion, then you can see the edit is cropped so you check the edited and original and click stack.

I started stacking edits when I switched to immich but I have a huge backlog of edited copies that probably look like duplicates.

**Comment by Str1atum** [comment]
Any changes in docker files necessary from last release?

**Reply by alextran1502** [reply]
If it is not mentioned in the release notes. Then you don't need to do anything besides pulling new images

**Reply by zackpollard** [reply]
> Any changes in docker files necessary from last release?

Nope, any changes will be listed at the top of the release notes if they are breaking/required 😊

**Comment by Str1atum** [comment]
Videos don't play any more since the update in Safari 17.5
Latest chrome works fine.

**Comment by oslo254804746** [comment]
Hello, have you encountered the following problems when upgrading from 1.106.4 to 1.107.1?

```
╰─ kubectl logs -f immich-server-64c9d9f895-4p8mj -n app                                                                                                                      ─╯
Detected CPU Cores: 8
Starting api worker
Starting microservices worker
[Nest] 7  - 07/03/2024, 1:33:53 AM     LOG [Microservices:EventRepository] Initialized websocket server
Migration "AddFaceSearchRelation1718486162779" failed, error: column "embedding" of relation "smart_search" does not exist
QueryFailedError: column "embedding" of relation "smart_search" does not exist
    at PostgresQueryRunner.query (/usr/src/app/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:219:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async AddFaceSearchRelation1718486162779.up (/usr/src/app/dist/migrations/1718486162779-AddFaceSearchRelation.js:17:9)
    at async MigrationExecutor.executePendingMigrations (/usr/src/app/node_modules/typeorm/migration/MigrationExecutor.js:225:17)
    at async DataSource.runMigrations (/usr/src/app/node_modules/typeorm/data-source/DataSource.js:265:35)
    at async DatabaseRepository.runMigrations (/usr/src/app/dist/repositories/database.repository.js:169:9)
    at async /usr/src/app/dist/services/database.service.js:134:17
    at async /usr/src/app/dist/repositories/database.repository.js:177:23 {
  query: 'ALTER TABLE smart_search ALTER COLUMN embedding SET STORAGE EXTERNAL',
  parameters: undefined,
  driverError: error: column "embedding" of relation "smart_search" does not exist
      at /usr/src/app/node_modules/pg/lib/client.js:526:17
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async PostgresQueryRunner.query (/usr/src/app/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:184:25)
      at async AddFaceSearchRelation1718486162779.up (/usr/src/app/dist/migrations/1718486162779-AddFaceSearchRelation.js:17:9)
      at async MigrationExecutor.executePendingMigrations (/usr/src/app/node_modules/typeorm/migration/MigrationExecutor.js:225:17)
      at async DataSource.runMigrations (/usr/src/app/node_modules/typeorm/data-source/DataSource.js:265:35)
      at async DatabaseRepository.runMigrations (/usr/src/app/dist/repositories/database.repository.js:169:9)
      at async /usr/src/app/dist/services/database.service.js:134:17
      at async /usr/src/app/dist/repositories/database.repository.js:177:23 {
    length: 125,
    severity: 'ERROR',
    code: '42703',
    detail: undefined,
    hint: undefined,
    position: undefined,
    internalPosition: undefined,
    internalQuery: undefined,
    where: undefined,
    schema: undefined,
    table: undefined,
    column: undefined,
    dataType: undefined,
    constraint: undefined,
    file: 'tablecmds.c',
    line: '8369',
    routine: 'ATExecSetStorage'
  },
  length: 125,
  severity: 'ERROR',
  code: '42703',
  detail: undefined,
  hint: undefined,
  position: undefined,
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'tablecmds.c',
  line: '8369',
  routine: 'ATExecSetStorage'
}
microservices worker exited with code 1
```

**Reply by oslo254804746** [reply]
Now I rolled back to 1.106.4, fortunately it works fine, I backed up my database immediately

**Reply by mertalev** [reply]
You seem to have dropped the `embedding` column for `smart_search` at some point (possibly when updating pgvecto.rs). You can re-add it by following the instructions [here](https://github.com/immich-app/immich/discussions/7310#discussioncomment-8559595) (in your case, the facial recognition data should be okay so you only need to do the parts related to `clip_index` and the `smart_search` table).

For others looking to run these queries, please make sure you back up your database first.

**Reply by oslo254804746** [reply]
thanks, it works, during execute the sql  add index may not work, need execute ` SELECT pgvectors_upgrade();` then restart `pgvecto.rs` pod , after restart pod add index worked

**Comment by IoSonoAndreaZ** [comment]
_all local assets will be displayed on the main timeline_

I really don't want this, how to avoid it?
On my phone I have many image folders with material not created by my camera and that I don't want to see in my timeline.

**Reply by JordyEGNL** [reply]
> I really don't want this, how to avoid it?
> On my phone I have many image folders with material not created by my camera and that I don't want to see in my timeline.

Update the mobile app to v1.107.1 or newer. The change is reverted for now :)

**Reply by IoSonoAndreaZ** [reply]
oh thanks!

**Comment by AndyPro720** [comment]
Got the following error(s) on 107.2 from 106, any ideas? Immich Server is unable to start. Thanks

```
immich_postgres          | 2024-07-04 06:28:19.238 UTC [165] ERROR:  internal error: entered unreachable code
immich_postgres          | 2024-07-04 06:28:19.238 UTC [165] STATEMENT:
immich_postgres          |                  CREATE INDEX face_index ON face_search
immich_postgres          |                  USING hnsw (embedding vector_cosine_ops)
immich_postgres          |                  WITH (ef_construction = 300, m = 16)
immich_server            | Migration "AddFaceSearchRelation1718486162779" failed, error: internal error: entered unreachable code
immich_server            | QueryFailedError: internal error: entered unreachable code
immich_server            |     at PostgresQueryRunner.query (/usr/src/app/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:219:19)
immich_server            |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
immich_server            |     at async AddFaceSearchRelation1718486162779.up (/usr/src/app/dist/migrations/1718486162779-AddFaceSearchRelation.js:23:9)
immich_server            |     at async MigrationExecutor.executePendingMigrations (/usr/src/app/node_modules/typeorm/migration/MigrationExecutor.js:225:17)
immich_server            |     at async DataSource.runMigrations (/usr/src/app/node_modules/typeorm/data-source/DataSource.js:265:35)
immich_server            |     at async DatabaseRepository.runMigrations (/usr/src/app/dist/repositories/database.repository.js:169:9)
immich_server            |     at async /usr/src/app/dist/services/database.service.js:134:17
immich_server            |     at async /usr/src/app/dist/repositories/database.repository.js:177:23 {
immich_server            |   query: '\n' +
immich_server            |     '            CREATE INDEX face_index ON face_search\n' +
immich_server            |     '            USING hnsw (embedding vector_cosine_ops)\n' +
immich_server            |     '            WITH (ef_construction = 300, m = 16)',
immich_server            |   parameters: undefined,
immich_server            |   driverError: error: internal error: entered unreachable code
immich_server            |       at /usr/src/app/node_modules/pg/lib/client.js:526:17
immich_server            |       at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
immich_server            |       at async PostgresQueryRunner.query (/usr/src/app/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:184:25)
immich_server            |       at async AddFaceSearchRelation1718486162779.up (/usr/src/app/dist/migrations/1718486162779-AddFaceSearchRelation.js:23:9)
immich_server            |       at async MigrationExecutor.executePendingMigrations (/usr/src/app/node_modules/typeorm/migration/MigrationExecutor.js:225:17)
immich_server            |       at async DataSource.runMigrations (/usr/src/app/node_modules/typeorm/data-source/DataSource.js:265:35)
immich_server            |       at async DatabaseRepository.runMigrations (/usr/src/app/dist/repositories/database.repository.js:169:9)
immich_server            |       at async /usr/src/app/dist/services/database.service.js:134:17
immich_server            |       at async /usr/src/app/dist/repositories/database.repository.js:177:23 {
immich_server            |     length: 82,
immich_server            |     severity: 'ERROR',
immich_server            |     code: 'XX000',
immich_server            |     detail: undefined,
immich_server            |     hint: undefined,
immich_server            |     position: undefined,
immich_server            |     internalPosition: undefined,
immich_server            |     internalQuery: undefined,
immich_server            |     where: undefined,
immich_server            |     schema: undefined,
immich_server            |     table: undefined,
immich_server            |     column: undefined,
immich_server            |     dataType: undefined,
immich_server            |     constraint: undefined,
immich_server            |     file: 'utils.rs',
immich_server            |     line: '21',
immich_server            |     routine: undefined
immich_server            |   },
immich_server            |   length: 82,
immich_server            |   severity: 'ERROR',
immich_server            |   code: 'XX000',
immich_server            |   detail: undefined,
immich_server            |   hint: undefined,
immich_server            |   position: undefined,
immich_server            |   internalPosition: undefined,
immich_server            |   internalQuery: undefined,
immich_server            |   where: undefined,
immich_server            |   schema: undefined,
immich_server            |   table: undefined,
immich_server            |   column: undefined,
immich_server            |   dataType: undefined,
immich_server            |   constraint: undefined,
immich_server            |   file: 'utils.rs',
immich_server            |   line: '21',
immich_server            |   routine: undefined
immich_server            | }
immich_server            | microservices worker exited with code 1
immich_postgres          | 2024-07-04 06:28:19.620 UTC [166] LOG:  unexpected EOF on client connection with an open transaction
immich_server exited with code 1

```

**Reply by bo0tzz** [reply]
https://github.com/immich-app/immich/issues/10767#issuecomment-2206506347

**Comment by jpaffrath** [comment]
Unfortunately for me the face detection got significantly worse with this update. My previous version showed me reliable detections of faces, now it "recognises" only on face and this is far from perfect, as there are many mismatched images detected. Am I the only one experiencing this? I ran the face detection job twice on all my assets.
